src/Hint/Context.hs

```diff
@@ -385,14 +380,27 @@ reset = do -- clean up context
            cleanPhantomModules
            --
            -- Now, install a support module
-           installSupportModule
+           res <- installSupportModule Nothing
+           mayFail (return $ guard (isSucceeded res) >> Just ())
 
 -- Load a phantom module with all the symbols from the prelude we need
-installSupportModule :: MonadInterpreter m => m ()
-installSupportModule = do mod <- addPhantomModule support_module
-                          onState (\st -> st{hintSupportModule = mod})
-                          mod' <- findModule (pmName mod)
-                          runGhc $ setContext [mod'] []
+installSupportModule :: MonadInterpreter m => Maybe [GHC.Target] -> m GHC.SuccessFlag
+installSupportModule tM = do mod <- addPhantomModule support_module
+                             onState (\st -> st{hintSupportModule = mod})
+                             case tM of
+                                Nothing -> do
+                                  mod' <- findModule (pmName mod)
+                                  runGhc $ setContext [mod'] []
+                                  return GHC.Succeeded
```

Comment:
I now understand that this `Nothing` case runs if we try and fail to load the modules which the user has requested. If the exception propagates all the way to the outside of the InterpreterT block, then it doesn't really matter what is in the context, but if the user catches the exception inside the InterpreterT block and then continues to evaluate code, it makes sense that we now want the context to be empty except for the support module.

Because this is such a rare corner case, I am worried about whether we have a test for that case. This comment is a reminder for me to check if we have a test case, and to add a new one otherwise.

Reply:
test added in af97170ad223ccca48b539484bfc442cf9641884